src/components/binding/binding.js

```diff
@@ -140,25 +149,66 @@ genotet.BindingView = function(viewName, params) {
     .on('genotet.addTrack', function() {
       var track = this.data.tracks.slice(-1).pop();
       this.loader.loadFullTrack(this.data.tracks.length, track.fileName,
-        this.data.chr);
+        this.data.chr, true);
     }.bind(this))
     .on('genotet.removeTrack', function(event, trackIndex) {
       this.data.tracks.splice(trackIndex, 1);
       this.renderer.render();
       this.panel.updateTracks();
     }.bind(this))
-    .on('genotet.gene', function(event, data) {
+    .on('genotet.updateTrack', function(event, data) {
       this.data.tracks[data.trackIndex].fileName = data.fileName;
       this.loader.loadFullTrack(data.trackIndex, data.fileName,
-        this.data.chr);
+        this.data.chr, false);
+    }.bind(this))
+    .on('genotet.loadBindingList', function() {
+      this.loader.loadBindingList();
+    }.bind(this))
+    .on('genotet.loadMapping', function(event, data) {
+      this.loader.loadMapping(data.mappingFileName, data.gene);
     }.bind(this));
 
   $(this.loader)
     .on('genotet.chr', function(event, chr) {
       this.panel.updateChr(chr);
     }.bind(this))
-    .on('genotet.track', function() {
+    .on('genotet.addPanelTrack', function() {
       this.panel.updateTracks();
+    }.bind(this))
+    .on('genotet.updateTrackList', function() {
+      this.panel.updateTrackList();
+    }.bind(this))
+    .on('genotet.updateLinkTrack', function(event, fileName) {
+      this.updateLinkTrack_(fileName);
+    }.bind(this));
+
+  // Set up link callbacks.
+  $(this)
+    .on('genotet.updateTrack', function(event, data) {
+      var genes = /** @type {!Array<string>} */ (data);
+
+      // Get the gene from the clicked node,
+      // or the source gene from the clicked edge.
+      var sourceGene = genes[0];
+      var mappingFileName = genotet.data.mappingFile['gene-binding'];
+      if (mappingFileName == 'Direct Mapping') {
+        var fileName = sourceGene + '.bw';
+        this.loader.signal('updateLinkTrack', fileName);
+      } else {
+        this.panel.signal('loadMapping', {
+          mappingFileName: mappingFileName,
+          gene: sourceGene
+        });
+      }
+    }.bind(this))
+    .on('genotet.locus', function(event, data) {
+      var genes = /** @type {!Array<string>} */ (data);
```

Comment:
add comments to describe that Array<string> contains 2 elements that are source and target genes of the clicked edge.
```js
/**
 * Comments go here.
 * @type {!Array<string>}
 */
var genes = /** @type {!Array<string>} */(data);
```